lib/libc/Kconfig

```diff
@@ -43,7 +43,7 @@ config NEWLIB_LIBC_SUPPORTED
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on !REQUIRES_FULL_LIBCPP || ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y")
```

Comment:
Let's figure out what this should look like.

For apps using the full C++ library, `PICOLIBC_SUPPORTED` depends on `TOOLCHAIN_HAS_PICOLIBC`. No other option is supported.

For all other apps, `PICOLIBC_SUPPORTED` depends on *either* `TOOLCHAIN_HAS_PICOLIBC` or `ZEPHYR_PICOLIBC_MODULE`.

Let's just make that right; I don't think we need to change anything other than this; once `PICOLIBC_SUPPORTED` is false, the existing libc selection logic will fall through to the minimal C library.

Here's the full logic table:
```
libc++  TOOLCHAIN       MODULE          PICOLIBC_SUPPORTED
  0         0             0                     0
  0         0             1                     1
  0         1             0                     1
  0         1             1                     1
  1         0             0                     0
  1         0             1                     0
  1         1             0                     1
  1         1             1                     1
```
By examination, we can see that
```
PICOLIBC_SUPPORTED = TOOLCHAIN || (MODULE && !libc++)
```
So, we can change the above depends line to:
```
depends on ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y") || (ZEPHYR_PICOLIBC_MODULE && !REQUIRES_FULL_LIBCPP)
```
Feel free to flip the disjunction around to make it look more like the original code.

Reply:
Updated the commit (&message)